pkg/autoscaler/autoscaler.go

```diff
@@ -105,7 +104,7 @@ func New(
 
 	return &Autoscaler{
 		namespace:       namespace,
-		revisionService: revisionService,
+		revisionService: deciderSpec.ServiceName, // To avoid locking to read the decider spec.
```

Comment:
You don't need to lock to read it or rather: it's already been read in each `Scale` call once (see `spec` in line 140). I'd get the ServiceName from that spec.